main.nf

```diff
@@ -2541,9 +3755,93 @@ process create_yaml {
 		set group, file("${group}.yaml*") into yaml
 
 	script:
+		"""
+		create_yml.pl \\
+			--g $group,$clarity_sample_id --d $diagnosis --panelsdef $params.panelsdef --out ${group}.yaml --ped $ped --files $INFO --assay $assay,$analysis --antype $params.antype
+		"""
 
-	"""
-	create_yml.pl \\
-		--g $group,$clarity_sample_id --d $diagnosis --panelsdef $params.panelsdef --out ${group}.yaml --ped $ped --files $INFO --assay $assay,$analysis --antype $params.antype
-	"""
+	stub:
+		"""
+		touch ${group}.yaml
+		"""
+}
+
+process combine_versions {
+	publishDir "${OUTDIR}/versions", mode: 'copy', overwrite: 'true', pattern: 'all_versions.yml'
+
+	input:
+		set group, versions from ch_fastp_versions.first().mix(
```

Comment:
Feedback on this approach would be appreciated. The "first" is there to prevent multiple identical versions being used when the process are ran in multiple instances, i.e. just pick one version blob from each.

Reply:
'first' is a great function to avoid multiple identical files and is indeed a smart way to handle concurrent instances

Reply:
OK, good to hear, thanks for the input!